### PR TITLE
quotas: add QuotaBase mixin

### PR DESCRIPTION
**Pull request by diegodelemos**
* Creates a quota related operations mixin to propagate quota
  calculation to different objects (workflows, interactive sessions...)
  (closes reanahub/reana-server#296).

### CODE REVIEW

File: reana_db/models.py [hunk 1]
```diff
@@ -511,38 +522,33 @@ def get_full_workflow_name(self):
         """Return full workflow name including run number."""
         return "{}.{}".format(self.name, str(self.run_number))
 
-    def get_workspace_disk_usage(self, summarize=False, block_size=None):
+    def get_workspace_disk_usage(self, summarize=False):
         """Retrieve disk usage information of a workspace."""
         from .database import Session
```
[the remark this answers is not in the document]
Comment (diegodelemos) [reply reana_db/models.py:527]: I think we do use it, [here](https://github.com/reanahub/reana-server/pull/301/files#diff-16a8293e6225f903f0412763851a5589f06f6f8891689da7abd0cde7a8166c52R2239). Unfortunately, we connot drop this method straight away, as we depend on the `du` output to do non summarized disk usage reports for workflow workspaces.

File: reana_db/models.py [hunk 2]
```diff
@@ -511,38 +522,33 @@ def get_full_workflow_name(self):
...
+        disk_usage = self.get_quota_usage().get("disk", {}).get("usage", {})
+        if not disk_usage:
+            # recalculate disk workflow resource
+            workflow_resource = store_workflow_disk_quota(self)
+            disk_usage = dict(
+                raw=workflow_resource.quota_used,
+                to_human_readable_units=ResourceUnit.human_readable_unit(
+                    ResourceUnit.bytes_, workflow_resource.quota_used
+                ),
+            )  # replace with another call `self.get_quota_usage().get("disk", {}).get("usage", {})`
```
[the remark this answers is not in the document]
Comment (diegodelemos) [reply reana_db/models.py:549]: Ups, it is a leftover, thanks for spotting.

File: reana_db/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mvidalgarcia) [reply reana_db/models.py:522]: I meant the import, sorry for not being specific 😀

File: reana_db/models.py [hunk 3]
```diff
@@ -53,7 +53,67 @@ def generate_uuid():
...
+        quota_limit = 0
+        unit = None
+
+        for resource in self.resources:
+            if resource.resource.type_ == resource_type:
+                # make sure that all resources of the same type use the same units
+                if unit and unit != resource.resource.unit:
+                    raise Exception(
+                        "Error while calculating quota usage. Not all "
+                        f"resources of resource type {resource_type} use "
```
Comment (mvidalgarcia) [inline reana_db/models.py:84]: Please convert to `format()`. Otherwise, `reana-client` tests might fail.

**Review by mvidalgarcia — APPROVED** [review]
Approving as I don't want to block it anymore and functionality-wise it works perfectly 🚀

**Pull request merged**